pytorch_lightning/core/datamodule.py

```diff
@@ -155,14 +234,14 @@ def prepare_data(self):
         """
 
     @abstractmethod
-    def setup(self, *args, **kwargs):
+    def setup(self, stage: Optional[str] = None):
```

Comment:
Added this to allow for people to set None as default value of `stage`, which will let them call `dm.setup()`. It will act as if you've setup both `fit` and `test` related setups if you do it this way, which won't confuse trainer when you go to call `trainer.test` after `trainer.fit`.

I documented this usage heavily in the docs.